crates/spark-wallet/src/wallet.rs

```diff
@@ -1227,6 +1224,19 @@ impl BackgroundProcessor {
                 .next()
```

Comment:
It would be nice if we could emit the pending payment event before querying the SSP. Or even not query the SSP at all. It's just metadata we're querying, which seems cacheable. Do you think we could cache that? (I didn't look at it long)

Reply:
My idea for the payment pending event was to notify immediately when hearing about the payment, so you can provide quick feedback in your UI. So I find the API call disturbing for that purpose. An api call adds time, so payment latency. I'm thinking maybe the details are not at all important for the PaymentPending event actually? Can we pass None instead? Or do 2 PaymentPending events in a row?

Reply:
I see what you mean Jesse but since spark operator gives us a payment with partial data I think it will be too confusing to emit it with partial data. Perhaps we can start this way (it is a nice progress over what we have now) and if we will see it is necessary we can add another event which emits quicker?

Reply:
Yes, not saying it is perfect. Agree that ideally we would push the pending even asap. On the other hand, I also feel it's important to provide consistent and correct payment data and we already make significant savings with the current approach as the heavist process is claiming the transfer.